tests/test_xray/test_xray_client.py

```diff
@@ -45,6 +45,37 @@ def test_xray_dynamo_request_id():
     setattr(requests.Session, "prepare_request", original_session_prep_request)
 
 
+def test_xray_dynamo_request_id_with_context_mgr():
+    with mock_xray_client():
```

Comment:
This test ensures that the context manager starts correctly, but I think it’s be a good idea to also add an assertion showing that it stops too.

Reply:
Good point - added